catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/ContentDirectoryMonitor.java

```diff
@@ -316,30 +324,99 @@ private String getBlackListAsRegex() {
     return String.join("|", patterns);
   }
 
-  public List<RouteDefinition> getRouteDefinitions() {
-    return camelContext.getRouteDefinitions();
+  /**
+   * This method acquires a lock on initialization and does:
```

Comment:
✏️`This method acquires a lock on initialization and:` and also mabe add some links for the classes below